UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Matrix.cs

```diff
@@ -33,21 +33,19 @@ private void Awake()
 		initialOffset = Vector3Int.CeilToInt(gameObject.transform.position);
 	}
 
+    public bool IsPassableAt(Vector3Int position)
+    {
+        return IsPassableAt(position + Vector3Int.up, position);
```

Comment:
Why Vector3Int.up? it's (0, 1, 0), so it will always check if one's approaching from the north, this will cause all kinds of quirks and inconsistencies

Reply:
Most contexts will have an origin, so IsPassableAt(origin, position) will be called and the wrapper method IsPassableAt(position) will not be called.
IsPassableAt(position) should be only called when the origin is irrelevant, therefore the origin argument does not matter as long as it is adjacent to the position. Vector3.Int.up/down/left/right should all return the same result in the contexts that the wrapper method IsPassableAt(position) is called.
One example of a call to IsPassableAt(position) is in the DoorController class. IsPassableAt(position) is called to see if a normal door (not a sliding door) is passable. Since normal doors can be passed from all directions, the origin argument is irrelevant as long as it is adjacent to the door position. Thus IsPassableAt(position) is called because the origin is not needed.

Reply:
still its behaviour would depend on whether tile above is solid, as `IsPassableAt(origin, target)` returns false if `origin` is solid and `target` is not.
Let's take ExplodeWhenShot for example, in your case it won't spawn fire tile if there's a wall/solid object on the tile above

Reply:
I did some tests and I can say that replacing this with
`return IsPassableAt(position, position);` certainly doesn't harm and has consistent behaviour regardless of surroundings